packages/patternfly-4/react-charts/src/components/ChartDonutUtilization/ChartDonutUtilization.js

```diff
@@ -0,0 +1,394 @@
+   * If given as an array of strings, or a string containing dots or brackets,
+   * it will be used as a nested object property path (for details see Lodash docs for _.get).
+   * If `null` or `undefined`, the data value will be used as is (identity function/pass-through).
+   * @example 0, 'x', 'x.value.nested.1.thing', 'x[2].also.nested', null, d => Math.sin(d)
+   */
+  x: PropTypes.string,
+  /**
+   * The y prop specifies how to access the Y value of each data point.
+   * If given as a function, it will be run on each data point, and returned value will be used.
+   * If given as an integer, it will be used as an array index for array-type data points.
+   * If given as a string, it will be used as a property key for object-type data points.
+   * If given as an array of strings, or a string containing dots or brackets,
+   * it will be used as a nested object property path (for details see Lodash docs for _.get).
+   * If `null` or `undefined`, the data value will be used as is (identity function/pass-through).
+   * @example 0, 'y', 'y.value.nested.1.thing', 'y[2].also.nested', null, d => Math.sin(d)
+   */
+  y: PropTypes.string
+};
+
+const ChartDonutUtilization = ({
+  data,
+  donutDx = 0,
+  donutDy = 0,
+  donutOrientation = 'left',
+  legendComponent,
+  legendData,
+  legendOrientation = styles.legend.orientation,
+  showStatic = true,
+  standalone = true,
+  subTitle,
+  themeColor,
+  themeVariant,
+  thresholds,
+  title,
+  transform,
+  x,
+  y,
+
+  // destructure last
+  theme = getDonutUtilizationTheme(themeColor, themeVariant),
+  donutHeight = theme.pie.height,
+  donutWidth = theme.pie.width,
+  height = theme.pie.height,
+  innerRadius = ((donutHeight || donutWidth) - 34) / 2,
+  width = theme.pie.width,
+  ...rest
+}) => {
+  // Returns computed data representing the pie chart percentage
+  const getComputedData = () => {
+    const datum = [{ ...data }];
+    const xVal = getDatumXVal(datum);
+    const yVal = getDatumYVal(datum);
+    const computedData = [{ x: xVal, y: yVal || 0 }];
+    if (showStatic) {
+      computedData.push({ y: yVal ? Math.abs(100 - yVal) : 100 });
+    }
+    return computedData;
+  };
+
+  // Returns a data point value via the x accessor
+  const getDatumXVal = datum => {
+    const result = map(datum, property(x || 'x'));
+    return result.length ? result[0] : undefined;
+  };
+
+  // Returns a data point value via the y accessor
+  const getDatumYVal = datum => {
+    const result = map(datum, property(y || 'y'));
+    return result.length ? result[0] || 0 : 0;
+  };
+
+  // Returns thresholds with default color scale
+  const getDonutThresholds = () => {
+    const result = [];
+    if (thresholds) {
+      const numColors = styles.thresholds.colorScale.length;
+      for (let i = 0; i < thresholds.length; i++) {
+        result.push({
+          color: thresholds[i].color ? thresholds[i].color : styles.thresholds.colorScale[i % numColors],
```

Comment:
Do we want to loop back through the threshold colors? This means they will just alternate between `#F0AB00` and `#C9190B`. I think it's probably okay for now since this captures the majority of use cases, but we might want to think about interpolating color values using something like https://gka.github.io/chroma.js/#color-scales

Reply:
Yes, those particular colors were chosen for their color contrast to support accessibility. (Ultimately, the colors will come from variables defined by PatternFly core.) Developers can add/override the colors if necessary, but they must ensure the colors are accessible. If we did use chroma.js, I don't know that we could guarantee the gradient would also be accessible?

Reply:
Yeah, I don't think we need to add chroma.js now. It's definitely an edge case and is easily overridden by providing your own colors, but if the user provided 3 thresholds and no colors, then when the last threshold is surpassed, the dynamic chart would change from red to yellow. That might be a little misleading.

Reply:
Understood.